tendermint/src/abci/response.rs

```diff
@@ -0,0 +1,1027 @@
+//! ABCI responses and response data.
+//!
+//! The [`Response`] enum records all possible ABCI responses. Responses that
+//! contain data are modeled as a separate struct, to avoid duplication of field
+//! definitions.
+
+// IMPORTANT NOTE ON DOCUMENTATION:
+//
+// The documentation for each response type is adapted from the ABCI Methods and
+// Types spec document. However, the same logical response may appear three
+// times, as a struct with the response data, as a Response variant, and as a
+// CategoryResponse variant. Ideally, the documentation would be copied between
+// these automatically, but doing this requires using #[doc = include_str!],
+// which is unstable. For now, the Response enum is the source of truth; please
+// change the docs there and copy as required.
+
+use std::convert::{TryFrom, TryInto};
+
+use bytes::Bytes;
+
+/// XXX(hdevalence): hide merkle::proof and re-export its contents from merkle?
+use crate::merkle::proof as merkle;
+
+use super::{
+    //Event,
+    event::Event,
+    params::ConsensusParams,
+    types::{Snapshot, ValidatorUpdate},
+};
+
+/// Returns an exception (undocumented, nondeterministic).
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Exception {
+    /// Undocumented.
+    pub error: String,
+}
+
+/// Returns a string sent in the request, to test an ABCI implementation.
+///
+/// [ABCI documentation](https://docs.tendermint.com/master/spec/abci/abci.html#echo)
+#[derive(Clone, PartialEq, Eq, Debug, Default)]
+pub struct Echo {
+    /// The message sent in the request.
+    pub message: String,
+}
+
+/// Returns information about the application state.
+///
+/// [ABCI documentation](https://docs.tendermint.com/master/spec/abci/abci.html#info)
+#[derive(Clone, PartialEq, Eq, Debug, Default)]
+pub struct Info {
+    /// Some arbitrary information.
+    pub data: String,
+    /// The application software semantic version.
+    pub version: String,
+    /// The application protocol version.
+    pub app_version: u64,
+    /// The latest block for which the app has called [`Commit`](super::request::Commit).
```

Comment:
For me, `cargo doc` complains that there's "no item named `Commit` in module `request`"

Reply:
Fixed